pkg/bmc/irmc.go

```diff
@@ -52,6 +52,9 @@ func (a *iRMCAccessDetails) DriverInfo(bmcCreds Credentials) map[string]interfac
 		"irmc_username": bmcCreds.Username,
 		"irmc_password": bmcCreds.Password,
 		"irmc_address":  a.hostname,
+		"ipmi_username": bmcCreds.Username,
+		"ipmi_password": bmcCreds.Password,
+		"ipmi_address":  a.hostname,
```

Comment:
In #848 there was a link to https://opendev.org/openstack/ironic/src/branch/master/ironic/drivers/modules/irmc/common.py#L160-L174 mentioned - do we know why Ironic isn't copying these fields, since it seems like that code should be copying the irmc credentials?

Reply:
The function is called from two places: the [iRMC power driver's `get_power_state`](https://opendev.org/openstack/ironic/src/branch/master/ironic/drivers/modules/irmc/power.py#L234-L246) and the [iRMC management driver's `validate`](https://opendev.org/openstack/ironic/src/branch/master/ironic/drivers/modules/irmc/management.py#L220-L232) methods.

#848 short-circuits the former by going directly to the ipmitool driver.

After reading the power state conductor persists the node to the DB, but after validating the management driver it looks like nothing gets persisted. That would explain @zhouhao3's findings that it worked when changing the power driver to ipmitool after initial creation, but not when changing it from the start.

In any event, this can't hurt.